sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -144,12 +170,34 @@ public String apply(String input) {
     PCollectionView<Map<DestinationT, String>> schemasView =
         inputInGlobalWindow.apply(new CalculateSchemas<>(dynamicDestinations));
 
+    TupleTag<WriteBundlesToFiles.Result<DestinationT>> writtenFilesTag =
+        new TupleTag<WriteBundlesToFiles.Result<DestinationT>>("writtenFiles"){};
+    TupleTag<KV<ShardedKey<DestinationT>, TableRow>> unwrittedRecordsTag =
+        new TupleTag<KV<ShardedKey<DestinationT>, TableRow>>("unwrittenRecords") {};
+    PCollectionTuple writeBundlesTuple = inputInGlobalWindow
+            .apply("WriteBundlesToFiles",
+                ParDo.of(new WriteBundlesToFiles<>(stepUuid, unwrittedRecordsTag,
```

Comment:
optional: consider instead making WriteBundlesToFiles a `PTransform` (instead of a ParDo), that would create these TupleTag's internally, set proper coders and so on, and its return type be a struct with getters `PCollection<...> getWrittenFiles()` and `getUnwrittenRecords()`